addons/sale/wizard/sale_order_discount.py

```diff
@@ -103,7 +103,7 @@ def _create_discount_lines(self):
                 self._prepare_discount_line_values(
                     product=discount_product,
                     amount=self.discount_amount,
-                    taxes=self.tax_ids,
+                    taxes=self.env['account.tax'],
```

Comment:
Is this really correct? So still no tax when fixed amount?

Reply:
I used the previous behaviour for this, but yes, I should also make it the same as in loyalty, and use that fixed amount to compute a percentage discount and apply accordingly. :facepalm: